orca-queue-redis/src/main/kotlin/com/netflix/spinnaker/orca/q/redis/RedisQueue.kt

```diff
@@ -83,7 +82,17 @@ class RedisQueue(
         ?.also { id ->
           val ack = this::ackMessage.partially1(id)
           redis.readMessage(id) { message ->
-            callback(message, ack)
+            val attempts = message.getAttribute(AttemptsAttribute::class.java)!!.attempts
```

Comment:
Yeah, `!!` is apparently designed so it looks deliberately ugly so you're encouraged to refactor such that you don't need it. Why can it potentially be null but we're certain here it won't be? I'd question whether we really need this kind of flexibility for the attributes where it's basically a typed map. Could we not have a `MessageAttributes` type with specific properties for the attributes we know we have? Do we anticipate adding some kind of flexible metadata there?